MapboxCoreNavigation/RouteController.swift

```diff
@@ -376,6 +376,24 @@ open class RouteController: NSObject {
 
         return CLLocation(coordinate: userCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: userCourse, speed: location.speed, timestamp: location.timestamp)
     }
+    
+    func closestStep(to coordinate: CLLocationCoordinate2D) -> (CLLocationDistance, Int)? {
+        var closestStep = Int.max
+        var lowestDistance = Double.infinity
```

Comment:
Use `nil` to indicate the absence of a closest step. That way the compiler forces you to account for a missing value instead of finding out when a ginormous value crashes the code. Also, combine `closestStep` and `lowestDistance` into a single tuple-typed variable, to ensure that both values are set simultaneously and don’t go out of sync.